Assets/Scripts/EnviornmentalFeatures/Hazards/LaserBeamHazard.cs

```diff
@@ -0,0 +1,136 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Unity.VisualScripting;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif // UNITY_EDITOR
+
+///-////////////////////////////////////////////////////////////////
+///
+public class LaserBeamHazard : MonoBehaviour
+{
+    [SerializeField] private GameObject pathPointPrefab;
+    [Space]
+    
+    [Header("Properties")]
+    [SerializeField] private bool gizmosOn = true;
+    [Tooltip("Enable to have laserbeam loop from last back to first path point")]
+    [SerializeField] private bool isLooping = true;
+    [Tooltip("Damage to blood")]
+    [SerializeField] private float damage = 5f;
+    [SerializeField] private float speed = 5f;
+
+    [SerializeField] private List<Transform> pathPoints = new List<Transform>();
+    private Transform laserBeam;
+
+    private Transform targetPathPoint;
+    private int pathPointIndex = 0;
+    private int pathPointDirection = 1; // Direction of the path points (-1 or 1)
+
+    ///-////////////////////////////////////////////////////////////////
+    ///
+    void Start()
+    {
+        laserBeam = transform.Find("LaserBeam");
```

Comment:
Small note, get this reference from the editor because Unity already has this null checking for you

Reply:
Fixed!